pages/books/[id].js

```diff
@@ -0,0 +1,58 @@
+import { useRouter } from "next/router";
+import books from "../../lib/books.json";
+import Header from "@/components/Header/Header";
+import Image from "next/image";
+import styled from "styled-components";
+
+const StyledImage = styled(Image)`
+  border-radius: 8px;
+  position: relative;
+  top: -20px;
+  box-shadow: 0 6px 20px 5px rgba(0, 0, 0, 0.19);
+`;
+const StyledBookDetail = styled.article`
+  border: 2px solid black;
+  border-radius: 8px;
+  width: 90%;
+  margin: 2rem auto;
+  text-align: center;
+`;
+const StyledSection = styled.section`
+  display: flex;
+  justify-content: space-around;
+  font-style: italic;
+`;
+const StyledDescription = styled.p`
+  text-align: justify;
+  padding: 15px;
+`;
+export default function BookDetailsPage() {
+  const router = useRouter();
+  const { id } = router.query;
+  const currentBook = books.find((book) => book.id === id);
+  if (!currentBook) {
+    return;
+  }
+
+  return (
+    <>
+      <Header />
+      <StyledBookDetail>
+        <StyledImage
+          src={currentBook.cover}
+          height={224}
+          width={150}
+          alt={`Cover Image of ${currentBook.title}`}
+        />
+        <h2>{currentBook.title}</h2>
+        <p>{currentBook.author}</p>
+        <StyledSection>
+          <span>{currentBook.genre}</span>
+          <span>{currentBook.publishYear}</span>
+          <span>{currentBook.pages} Pages</span>
+        </StyledSection>
```

Comment:
Converting the StyledSection to use a `<ul>` with `<li>` elements could be an improvement to provide clearer semantics which is beneficial for screen readers.

> 💡Consider defining the base styling for `<ul>` elements in global styles to avoid the frustrations caused by default list styling. Typically, `<ul>` and `<ol>` have left padding to accommodate markers, affecting `<li>` items.


```suggestion
        <StyledDescriptionList>
          <li>{currentBook.genre}</li>
          <li>{currentBook.publishYear}</li>
          <li>{currentBook.pages} Pages</li>
        </StyledDescriptionList>
```